Yoast/javascript
# Add readability checks for Indonesian

## Summary


This PR can be summarized in the following changelog entry:

* [yoastseo] N/A Adds specs for the sentence length assessment in Indonesian.
* [yoastseo] Adds Sentence beginning assessment for Indonesian.
* [yoastseo] Adds Transition words assessment for Indonesian.
* [yoastseo] Adds Passive voice assessment for Indonesian. 
* [Yoast SEO Free] Adds the Sentence beginnings assessment for Indonesian.
* [Yoast SEO Free] Adds Transition words assessment for Indonesian.
* [Yoast Seo Free] Adds Passive voice assessment for Indonesian.

## Relevant technical choices:

* No user-facing changelog was added for the sentence length assessment since the functionality is already present and we just confirmed that the default value of 20 words is a good value for Indonesian. 

## Test instructions

This PR can be tested by following these steps:
* Run `yarn test` and make sure all tests pass.

* In your WordPress SEO Free testing environment, set the language to `Bahasa Indonesia`.
* Open a new post or page.
* For all the following assessments check the Readability section (`Keterbacaan`)
* Test the sentence length assessment (`Panjang kalimat`):
     * Write four or five Indonesian sentences that have more than 20 words per sentence (E.g., `Habib Muhammad bin Yahya lahir pada tahun 1260 Hijriyah (1844 Masehi) di Al-Masilah, sebuah desa kecil di Hadramaut dan ia tinggal di kota Mekah selama masa mudanya.`). The sentence length assessment should display this message: `100% of the sentences contain more than 20 words, which is more than the recommended maximum of 25%. 
     * Try to shorten the sentences. Write four Indonesian sentences that have 20 words or less per sentence (E.g., `Sejarah Indonesia banyak dipengaruhi oleh bangsa lainnya.`). The sentence length assessment should display this message: `Great!`
* Test the sentence beginning assessment:
     * Write three or more Indonesian sentences that start with different words. The sentence beginning readability assessment should read: `Consecutive sentences: There is enough variety in your sentences. That's great!`
     * Write three or more Indonesian sentences that begin with the same word for which the sentence beginning readability assessment should read: `Consecutive sentences: The text contains 3 consecutive sentences starting with the same word (E.g. `Rumahnya mewah. Rumahnya nyaman. Rumahnya mahal.`). Try to mix things up!`
* Test the transition words assessment:
     * Add a sentence containing a single-word transition word, like `selanjutnya`, `ringkasnya`, `karenanya` (E.g. `Ringkasnya, ia telah menyelesaikan pendidikannya dengan gemilang.`, `Selanjutnya, ia berencana untuk tinggal di Australia.`, `Karenanya, ia telah menjadi pengusaha sukses sekarang.`).
     * Check that the assessment bullet turns green and the transition words are correctly highlighted.
     * Add a sentence containing a multi-word transition word, like `bagaimanapun juga`, `dalam hal itu`, `pada dasarnya` (E.g. `Bagaimanapun juga, ibunya adalah orang yang paling berjasa dalam hidupnya.`, `Dalam hal itu, ia lebih memilih untuk tidak ikut campur.`, `Pada dasarnya, manusia adalah makhluk sosial`).
     * Check that the assessment bullet turns green and the transition words are correctly highlighted.
     * Remove the transition words from the sentences and check that the bullet turns red.
* Test the passive voice assessment:
     * Add these sentences to your post:
          `Penumpang itu dilempar ke luar`,
          `Sudahkah anak merasa didengarkan?`,
          `Suratnya belum dikirimkan ke alamat yang benar`.
     * Check that the passive voice bullet (`Kalimat pasif`) turns red.
     * Add the word `untuk` to every sentence, before `dilempar`, `didengarkan`, `dikirimkan` 
     * Check that the passive voice bullet turns green.

## Impact check

This PR affects the following parts of the plugin, which may require extra testing:
* Detection of passive voice in Dutch where the past participle form is followed by punctuation marks, such as: `!?,.")>']`

## UI changes

## Quality assurance


Fixes https://yoast.atlassian.net/browse/LIN-466
